codec-http2/src/main/java/io/netty/handler/codec/http2/Http2LocalFlowController.java

```diff
@@ -66,4 +66,50 @@ void receiveFlowControlledFrame(ChannelHandlerContext ctx, Http2Stream stream, B
      * @return the number of unconsumed bytes for the stream.
      */
     int unconsumedBytes(Http2Stream stream);
+
+    /**
+     * Factory method to generate a new {@link LocalFlowState} object. This will be called after a new stream object
+     * is created and before {@link Http2Connection.Listener#onStreamAdded(Http2Stream)} is called. The result of this
+     * method will be returned by {@link Http2Stream#localFlowState()} for use by {@link Http2LocalFlowController}.
+     * @param stream The stream to associate the {@link LocalFlowState} with.
+     * This may be any stream, including the {@link Http2Connection#connectionStream()}.
+     * @return A new object which will represent local flow control state for {@code stream}.
+     */
+    LocalFlowState newLocalFlowState(Http2Stream stream);
```

Comment:
@nmittler - I think there is value in having both, and within the logical responsibilities of flow control.  For example the user may want to take action (send window_update) if the `unavailableWindow()` drops below a certain threshold.  Also the unavailable and available may not be reasonable to calculate based upon `initial window size` (if the user was so inclined to track this) because we could go negative.